### PR TITLE
Allow source_date_epoch to be set from Config

### PR DESCRIPTION
**Pull request by mobileoverlord**
This will set `SOURCE_DATE_EPOCH` to a value from `config :nerves, source_date_epoch: "1234"` or can be overridden if `SOURCE_DATE_EPOCH` is already defined.

### CODE REVIEW

File: lib/nerves/env.ex [hunk 1]
```diff
@@ -386,6 +387,21 @@ defmodule Nerves.Env do
...
+      System.put_env("SOURCE_DATE_EPOCH", source_date_epoch)
+    end
+  end
+
+  defp source_date_epoch() do
+    (System.get_env("SOURCE_DATE_EPOCH") || Application.get_env(:nerves, :source_date_epoch))
+    |> case do
+      nil -> nil
+      int when is_integer(int) -> Integer.to_string(int)
+      str when is_binary(str) -> str
```
Comment (fhunleth) [inline lib/nerves/env.ex]: Could we have a check on the string to make sure that it's an integer? Specifically, I'd like to protect against someone setting `SOURCE_DATE_EPOCH` to the empty string and getting 1970 through out. I may have made that mistake.

**Comment by fhunleth** [comment]
I like it. I'm pretty sure that I'd enabled this on every project I make.

**Comment by mobileoverlord** [comment]
I was going to add this to the new project generator once we get it in.

File: lib/nerves/env.ex (hunk 1 above)
Comment (mobileoverlord) [reply lib/nerves/env.ex]: I updated this to validate the input

**Pull request merged**